examples/NavigationSample.elm

```diff
@@ -0,0 +1,85 @@
+module NavigationSample exposing (program)
+
+import Html exposing (..)
+import Html.Attributes exposing (class)
+import Html.Events exposing (..)
+import Navigation
+
+
+program : Program Never Model Msg
+program =
+    Navigation.program UrlChange
+        { init = init
+        , view = view
+        , update = update
+        , subscriptions = \_ -> Sub.none
+        }
+
+
+
+-- MODEL
+
+
+type alias Model =
+    { history : List Navigation.Location
+    }
+
+
+init : Navigation.Location -> ( Model, Cmd Msg )
+init location =
+    ( Model [ location ]
+    , Cmd.none
+    )
+
+
+
+-- UPDATE
+
+
+type Msg
+    = UrlChange Navigation.Location
+    | Go String
+
+
+
+{- We are just storing the location in our history in this example, but
+   normally, you would use a package like evancz/url-parser to parse the path
+   or hash into nicely structured Elm values.
```

Comment:
Doesn't necessarily need to be done for this PR, but I think it would be good to make this example a bit more realistic and to the URL parsing.